### PR TITLE
fix: show all events if there isnt BUY events

### LINKED ISSUE #2621: Empty history in gallery item
Since we set "BUY" event by default in #2567, some nft show empty data when loaded
Tested on: `/rmrk/gallery/11220191-daf9e25824d4bd9357-KUBOTS-KUBOT_459-0000000000000459`

#### screenshot
Screenshot 2022-03-21 at 14-21-29 KuBot #459

#### solution
- set "all" if there isn't "BUY" events

### PR DESCRIPTION
**Pull request by Jarsen136**
**Thank you for your contribution** to the [KodaDot NFT gallery](https://kodadot.xyz).
👇 \_ Let's make a quick check before the contribution.

### PR type

- [x] Bugfix
- [ ] Feature
- [ ] Refactoring

### What's new?

- [x] PR closes #2621 
- [ ] <brief_description_of_what_I've_added>

### Before submitting Pull Request, please make sure:

- [x] My contribution builds **clean without any errors or warnings**
- [x] I've merged recent default branch -- **main** and I've no conflicts
- [x] I've tried to respect high code quality standards
- [x] I've didn't break any original functionality
- [x] I've posted a screenshot of demonstrated change in this PR

### Optional

- [ ] I've tested it at </rmrk/collection/26902bc2f7c20c546a-1FVG7>
- [ ] I've tested PR on mobile and everything seems works
- [ ] I found edge cases
- [ ] I've written some unit tests 🧪

### Had issue bounty label?

- [x] Fill up your KSM address: [Payout](https://beta.kodadot.xyz/transfer/?target=Caiv9TbPz68q5dC8EcHu5xKYPRnremimGzqmEejDFNpWWLG)

### Community participation

- [x] [Are you at KodaDot Discord?](https://discord.gg/35hzy2dXXh)

### Screenshot

- [x] My fix has changed **something** on UI; a screenshot is best to understand changes for others.

image

### CODE REVIEW

File: components/rmrk/Gallery/History.vue [hunk 1]
```diff
@@ -312,6 +312,11 @@ export default class History extends mixins(ChainMixin, KeyboardEventsMixin) {
 
     this.data = this.data.reverse()
     this.filterData()
+
+    if (this.data.length === 0) {
```
Comment (roiLeo) [inline components/rmrk/Gallery/History.vue:316]:
```suggestion
    if (!this.data.length) {
```

Comment (roiLeo) [reply components/rmrk/Gallery/History.vue:316]: personal preference

**Review by roiLeo — COMMENTED** [review]
yep it does the job

File: components/rmrk/Gallery/History.vue (hunk 1 above)
Comment (Jarsen136) [reply components/rmrk/Gallery/History.vue]: oki

**Review by roiLeo — APPROVED** [review]

**Comment by yangwao** [comment]
pay 100 usd

**Comment by yangwao** [comment]
😍 Perfect, I’ve sent the payout 
💵 $100 @ 158.06 USD/KSM ~ 0.633 $KSM
🧗 [Caiv9TbPz68q5dC8EcHu5xKYPRnremimGzqmEejDFNpWWLG](https://kusama.subscan.io/account/Caiv9TbPz68q5dC8EcHu5xKYPRnremimGzqmEejDFNpWWLG)
🔗 [0x6c402843d80118150ff6ad43b0087b1cabacc54883f60a42d7f3f38a0349419c](https://kusama.subscan.io/extrinsic/0x6c402843d80118150ff6ad43b0087b1cabacc54883f60a42d7f3f38a0349419c)

🪅 Let’s grab another issue and get rewarded! 
🪄 [github.com/kodadot/nft-gallery/issues](https://github.com/kodadot/nft-gallery/issues/)

**Pull request merged**